src/sentry/static/sentry/app/components/forms/selectAutocompleteControl.jsx

```diff
@@ -0,0 +1,123 @@
+import {debounce} from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {t} from 'app/locale';
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {Client} from 'app/api';
+import handleXhrErrorResponse from 'app/utils/handleXhrErrorResponse';
+
+import SelectControl from './selectControl';
+
+class SelectAutocompleteControl extends React.Component {
+  static propTypes = {
+    /**
+     * API endpoint URL
+     */
+    url: PropTypes.string.isRequired,
+
+    /**
+     * Parses the results of API call for the select component
+     */
+    onResults: PropTypes.func,
+
+    /**
+     * Additional query parameters when sending API request
+     */
+    onQuery: PropTypes.func,
+
+    value: PropTypes.any,
+  };
+
+  static defaultProps = {
+    placeholder: '--',
+  };
+
+  constructor(props) {
+    super(props);
+    this.api = new Client();
+    this.state = {
+      query: '',
+    };
+    this.cache = {};
+  }
+
+  componentWillUnmount() {
+    if (!this.api) return;
+    this.api.clear();
+    this.api = null;
+  }
+
+  doQuery = debounce(cb => {
```

Comment:
this is definitely a nitpick, but wouldn't throttle be technically more correct

Reply:
I think it depends on what your desired behavior is:

* `throttle` means call this function at most once in **period**
* `debounce` means call this function after **period** passes since the last attempted call

I think when handling user input, it's fine to wait until the user is done typing before performing the request.